File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, nums, target):
        left, right = 0, len(nums) - 1
        while left < right:
            if nums[left] + nums[right] > target:
                right -= 1
            elif nums[left] + nums[right] < target:
                left += 1
            else:
                return [left+1, right+1]

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        // 오름차순 배열 numbers
        int ptr = numbers.length - 1;
        int idx = 0;
        

        while (idx < ptr) {
            int sum = numbers[idx] + numbers[ptr];
            if (sum == target)
                return new int[] {idx + 1, ptr + 1};
            else if (sum > target) {
                ptr--;
            }
            else {
                idx++;
            }
        }
        return null;
    }
}